mesonbuild/compilers/detect.py

```diff
@@ -1164,6 +1167,31 @@ def detect_swift_compiler(env: 'Environment', for_machine: MachineChoice) -> Com
 
     raise EnvironmentException('Unknown compiler "' + ' '.join(exelist) + '"')
 
+def detect_asm_compiler(env: 'Environment', for_machine: MachineChoice) -> Compiler:
+    compilers, _, _ = _get_compilers(env, 'asm', for_machine)
+    is_cross = env.is_cross_build(for_machine)
+
+    # We need a C compiler to properly detect the machine info
+    cc = detect_c_compiler(env, for_machine)
+    from ..environment import detect_machine_info
+    info = detect_machine_info({'c': cc})
+
+    popen_exceptions: T.Dict[str, Exception] = {}
+    for comp in compilers:
+        try:
+            output = Popen_safe(comp + ['-v'])[1]
+        except OSError as e:
+            popen_exceptions[' '.join(comp + ['-v'])] = e
+            continue
+
+        version = search_version(output)
+        if 'NASM' in output:
+            comp_class = NasmCompiler
+            env.coredata.add_lang_args(comp_class.language, comp_class, for_machine, env)
+            return comp_class(comp, version, for_machine, info, cc.linker, is_cross=is_cross)
+    _handle_exceptions(popen_exceptions, compilers)
+    raise EnvironmentException('Unreachable code (exception to make mypy happy)')
```

Comment:
this shouldn't be required, mypy konws that `_handle_exceptions` is `NoReturn` and should be happy with that.

Reply:
done.

Reply:
Sorry, no, mypy is not that smart: `mesonbuild/compilers/detect.py:666:1: error: Missing return statement`